### PR TITLE
Ux anchor upgrade

### PR DESCRIPTION
**Pull request by ivanwills**
Tests still to come

### CODE REVIEW

File: src/components/ux-anchor/ux-anchor.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+				var customEventName = customParams[0];
+				if (customParams.length > 1) {
+					var params = customParams[1].split(',');
+					this.fire(customEventName, this, params);
+				} else {
+					this.fire(customEvent, this);
+				}
+				return false;
+			} else {
+				console.log('default event fired');
```
Comment (nathf) [inline src/components/ux-anchor/ux-anchor.js]: Remove log

File: src/components/ux-anchor/ux-anchor.js [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+	},
+	onconfig: function () {
+		this.on('anchorClicked', function () {
+			var customEvent = this.get('ontap');
+			if (customEvent) {
+				var customParams = customEvent.split(':');
+				var customEventName = customParams[0];
+				if (customParams.length > 1) {
+					var params = customParams[1].split(',');
+					this.fire(customEventName, this, params);
```
Comment (nathf) [inline src/components/ux-anchor/ux-anchor.js]: Do we want to call/apply these params or pass as an array?

Maybe something like this?
`.apply(this, [customEventName, this].concat(params))`

File: src/components/ux-anchor/README.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
 > Create anchors in your application
 
 `&lt;ux-anchor&gt;` is an internal helper component created for ractive-foundation.
+
+It will throw custom events (with custom data) by the use of the attribute `customEvent` e.g.
+
+`&lt;ux-anchor customEvent="myEvent:myData1,myData2" … &gt;content&lt;/ux-anchor&gt;`
```
Comment (nathf) [inline src/components/ux-anchor/README.md]: `customEvent` should be `ontap`

File: src/components/ux-anchor/manifest.json [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
+		"tabindex": "Set the tab index",
 		"target": "Set the target of the link",
-		"title": "Add a title to the link",
-		"content": "The content of the link if isDataModel is used",
-		"tabindex": "Set the tab index"
+		"title": "Add a title to the link"
 	},
 	"content": "Content of ux-anchor is yielded to if not in isDataModel mode when content is used",
 	"events": {
+		"anchorClicked": "Event to fire when component is clicked"
```
Comment (nathf) [inline src/components/ux-anchor/manifest.json:15]: ontap here too

Comment (ivanwills) [inline src/components/ux-anchor/manifest.json:15]: This event is thrown, but above was a `customEvent` reference which I have fixed

**Pull request merged**